<launch>
    <arg name="vehicle_launch" default="suav" />
    <param name="use_sim_time" type="bool" value= "false" />
    <node pkg="rtls_uwb_sensor" type="display" name="$(arg vehicle_launch)_display" output="log" >
        <param name="rtls_serial_port"  value="/dev/ttyUSB2" />
        <remap from="/uwb/data" to="/$(arg vehicle_launch)/uwb/data" />
    </node>
    <node pkg="range_localization" type="range_node" name="$(arg vehicle_launch)_localization" output="screen" >
        <rosparam command="load" file="$(find range_localization)/config/uav.yaml" />
        <rosparam command="load" file="$(find range_localization)/config/anchor_position_nooploop.yaml" />
        <param name="vehicle_name"  value="$(arg vehicle_launch)" />
        
        <param name="uwb_sensor_type"  value="rtls" />
        <!-- <param name="uwb_sensor_type"  value="nooploop" /> -->

        <param name="use_imu"  value="false" />
        <param name="imu_topic" value="/$(arg vehicle_launch)/dji_osdk_ros/imu" />

        <param name="range_topic"  value="/$(arg vehicle_launch)/uwb/data" />

        <param name="use_pressure_height"  value="false" />
        <param name="pressure_topic"  value="/$(arg vehicle_launch)/dji_osdk_ros/barometer_height" />
        <param name="motor_speed_topic"  value="/$(arg vehicle_launch)/dji_osdk_ros/motor_speed" />
        
        <param name="use_laser_height"  value="false" />
        <param name="height_topic"  value="/$(arg vehicle_launch)/dji_osdk_ros/height_above_takeoff" />

        <rosparam param="init_position" >[0.0, 0.0, 0.0]</rosparam>
        <rosparam param="init_gps_point" >[39.6530266923, 116.098553547, 27.8607826233]</rosparam>
    </node>
    <arg name="rviz" default="false" />
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find range_localization)/rviz_cfg/uwb_loc.rviz" />
    </group>
    <!-- <node pkg="rosbag" type="record" name="suav_2_bag_record" args="/suav_2/uwb/data /suav_2/imu/data /suav_2/filter/odom /suav_2/dji_osdk_ros/height_above_takeoff /suav_2/dji_osdk_ros/vo_position /suav_2/dji_osdk_ros/imu -o /home/$(env USER)/suav_2.bag"/> -->
</launch>